pkg/test/framework/components/echo/common/deployment/namespace.go

```diff
@@ -70,122 +70,48 @@ type EchoNamespace struct {
 }
 
 func (n EchoNamespace) build(t resource.Context, b deployment.Builder, cfg Config) deployment.Builder {
-	b = b.WithConfig(echo.Config{
-		Service:         ASvc,
-		Namespace:       n.Namespace,
-		ServiceAccount:  true,
-		Ports:           ports.All(),
-		Subsets:         []echo.SubsetConfig{{}},
-		Locality:        "region.zone.subzone",
-		IncludeExtAuthz: cfg.IncludeExtAuthz,
-	}).
-		WithConfig(echo.Config{
-			Service:         BSvc,
-			Namespace:       n.Namespace,
-			ServiceAccount:  true,
-			Ports:           ports.All(),
-			Subsets:         []echo.SubsetConfig{{}},
-			IncludeExtAuthz: cfg.IncludeExtAuthz,
-		}).
-		WithConfig(echo.Config{
-			Service:         CSvc,
-			Namespace:       n.Namespace,
-			ServiceAccount:  true,
-			Ports:           ports.All(),
-			Subsets:         []echo.SubsetConfig{{}},
-			IncludeExtAuthz: cfg.IncludeExtAuthz,
-		}).
-		WithConfig(echo.Config{
-			Service:         HeadlessSvc,
-			Namespace:       n.Namespace,
-			ServiceAccount:  true,
-			Headless:        true,
-			Ports:           ports.Headless(),
-			Subsets:         []echo.SubsetConfig{{}},
-			IncludeExtAuthz: cfg.IncludeExtAuthz,
-		}).
-		WithConfig(echo.Config{
-			Service:         StatefulSetSvc,
-			Namespace:       n.Namespace,
-			ServiceAccount:  true,
-			Headless:        true,
-			StatefulSet:     true,
-			Ports:           ports.Headless(),
-			Subsets:         []echo.SubsetConfig{{}},
-			IncludeExtAuthz: cfg.IncludeExtAuthz,
-		}).
-		WithConfig(echo.Config{
-			Service:        NakedSvc,
-			Namespace:      n.Namespace,
-			ServiceAccount: true,
-			Ports:          ports.All(),
-			Subsets: []echo.SubsetConfig{
-				{
-					Annotations: map[echo.Annotation]*echo.AnnotationValue{
-						echo.SidecarInject: {
-							Value: strconv.FormatBool(false),
-						},
-					},
-				},
-			},
-		}).
-		WithConfig(echo.Config{
-			Service:        TproxySvc,
-			Namespace:      n.Namespace,
-			ServiceAccount: true,
-			Ports:          ports.All(),
-			Subsets: []echo.SubsetConfig{{
-				Annotations: echo.NewAnnotations().Set(echo.SidecarInterceptionMode, "TPROXY"),
-			}},
-			IncludeExtAuthz: cfg.IncludeExtAuthz,
-		}).
-		WithConfig(echo.Config{
-			Service:         VMSvc,
-			Namespace:       n.Namespace,
-			ServiceAccount:  true,
-			Ports:           ports.All(),
-			DeployAsVM:      true,
-			AutoRegisterVM:  true,
-			Subsets:         []echo.SubsetConfig{{}},
-			IncludeExtAuthz: cfg.IncludeExtAuthz,
-		})
+	if cfg.Custom == nil {
+		defaultConfigs := CreateDefaultConfig(n.Namespace, cfg)
```

Comment:
A few things here:

1. Maybe just call it `DefaultEchoConfigs`
2. Instead of having this logic here, just add a method to the config `cfg.getConfigs` or something. It just returns `cfg.Configs` if not null, otherwise returns `DefaultConfigs()`.
3. Just have all the default configs have an empty namespace.  The logic should interpret an empty namespace to mean: deploy to all namespaces.